### PR TITLE
generator and python library improvements

### PR DESCRIPTION
**Pull request by jck**
/cc @mookerji @denniszollo

### CODE REVIEW

File: generator/sbpg/targets/python.py [hunk 1]
```diff
@@ -58,24 +58,32 @@ def construct_format(f, type_map=CONSTRUCT_CODE):
...
+    # return "%s('%s')" % (type_map.get(f.type_id), f.identifier)
   elif f.type_id == 'string' and f.options.get('size', None):
-    return "String('%s', %d)" % (f.identifier, f.options['size'].value)
+    return "'{id}'/ String({size}, paddir='left')".format(id=f.identifier,
+                                                   size=f.options['size'].value)
+    # return "String('%s', %d)" % (f.identifier, f.options['size'].value)
   elif f.type_id == 'string':
-    return "greedy_string('%s')" % (f.identifier)
+    return "'{id}' / GreedyString(encoding='utf8')".format(id=f.identifier)
+    # return "greedy_string('%s')" % (f.identifier)
```
Comment (mookerji) [inline generator/sbpg/targets/python.py]: Remove this comment

File: generator/sbpg/targets/python.py [hunk 2]
```diff
@@ -58,24 +58,32 @@ def construct_format(f, type_map=CONSTRUCT_CODE):
...
   elif f.type_id == 'array' and f.options.get('size', None):
     fill = f.options['fill'].value
     f_ = copy.copy(f)
     f_.type_id = fill
     s = f.options.get('size', None).value
-    return "Struct('%s', Array(%d, %s))" % (f.identifier, s, construct_format(f_))
+    # return "Struct(Array(%d, %s))" % (f.identifier, s, construct_format(f_))
```
Comment (mookerji) [inline generator/sbpg/targets/python.py]: Remove this comment

File: generator/sbpg/targets/python.py [hunk 3]
```diff
@@ -58,24 +58,32 @@ def construct_format(f, type_map=CONSTRUCT_CODE):
...
   elif f.type_id == 'array':
     fill = f.options['fill'].value
     f_ = copy.copy(f)
     f_.type_id = fill
-    return "OptionalGreedyRange(%s)" % construct_format(f_)
+    return "GreedyRange(%s)" % construct_format(f_)
   else:
-    return "Struct('%s', %s._parser)" % (f.identifier, f.type_id)
+    return "'%s' / Struct(%s._parser)" % (f.identifier, f.type_id)
+    # return "Struct('%s', %s._parser)" % (f.identifier, f.type_id)
```
Comment (mookerji) [inline generator/sbpg/targets/python.py]: Remove this comment

File: python/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
-construct==2.5.2
+construct==2.8.12
```
Comment (mookerji) [inline python/requirements.txt]: We may need a check in the library somewhere that asserts if this version is incorrect. People don't always update their dependencies.

File: python/sbp/acquisition.py [hunk 5]
```diff
@@ -60,19 +60,20 @@ class AcqSvProfile(object):
...
-                     ULInt8('int_time'),
-                     Struct('sid', GnssSignal._parser),
-                     ULInt16('bin_width'),
-                     ULInt32('timestamp'),
-                     ULInt32('time_spent'),
-                     SLInt32('cf_min'),
-                     SLInt32('cf_max'),
-                     SLInt32('cf'),
-                     ULInt32('cp'),))
+  # _parser = Embedded(Struct("AcqSvProfile",
```
Comment (mookerji) [inline python/sbp/acquisition.py]: Remove this comment

Comment (mookerji) [inline python/sbp/acquisition.py]: and all the ones below

File: python/sbp/utils.py [hunk 6]
```diff
@@ -78,7 +78,8 @@ def fmt_repr(obj):
 # TODO: Once https://github.com/construct/construct/pull/59 is published
 #       to pypi, remove this in favor of construct.GreedyString
 def greedy_string(name):
```
Comment (mookerji) [inline python/sbp/utils.py]: I would recommend having an assertion here as opposed to failing silently.

Comment (mookerji) [inline python/sbp/utils.py]: Or removing it?

**Comment by mookerji** [comment]
@jck see comments about comments. 

Have you tried using this with gnss-analysis and the console yet?

(This is *great*.)

**Comment by jck** [comment]
Yes, I've tried with both! and it does not seem to break.

File: python/requirements.txt (hunk 4 above)
Comment (jck) [inline python/requirements.txt]: setup.py reads this file into install_requires, so python will not allow sbp to be installed without this version of construct.

Comment (mookerji) [inline python/requirements.txt]: @jck excellent!

**Comment by mookerji** [comment]
@jck sweet.

**Comment by jck** [comment]
Forgot to mention, console needs changes since it has a bunch of star imports which clash with new construct variables.

**Comment by mookerji** [comment]
:shipit:

**Comment by mookerji** [comment]
Let's plan on cutting a separate release for gnss-analysis later this week.

**Pull request merged**